.github/code/nipoppy_track_processing_for_dataset.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+set -euo pipefail
+
+DS_ID=$1
+DS_URL="https://github.com/OpenNeuroDatasets/${DS_ID}.git"
+
+SCRIPT_DIR="$(dirname "$(realpath "$0")")"
+DERIVATIVES_REPO_MAP="derivatives/repo_derivatives_map.json"
+SOURCE_WORKDIR="derivatives/source"
+NIPOPPY_WORKDIR="derivatives/nipoppy"
+TEMPLATE_CONFIGS_DIR="derivatives/template_pipeline_configs"
```

Comment:
would make sense to add these dirs to a .gitignore inside `derivatives` to avoid accidentally committing a ton of stuff, e.g. when running locally